### DISCUSSION TITLE
LayVim on Windows wezterm terminal occrur error

### DISCUSSION THREAD

**Discussion by Create-Peace** [LazyVim/LazyVim#4659] [General]
Error in decoration provider treesitter/highlighter.win:
Error executing lua: ...s/Neovim/share/nvim/runtime/lua/vim/treesitter/query.lua:252: Query error at 52:2. Invalid node type "function_expression":
(function_expression
 ^

stack traceback:
	[C]: in function '_ts_parse_query'
	...s/Neovim/share/nvim/runtime/lua/vim/treesitter/query.lua:252: in function 'fn'
	...iles/Neovim/share/nvim/runtime/lua/vim/func/_memoize.lua:58: in function 'fn'
	...iles/Neovim/share/nvim/runtime/lua/vim/func/_memoize.lua:58: in function 'get'
	...im/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:28: in function 'new'
	...im/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:240: in function 'get_query'
	...im/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:188: in function 'fn'
	...m/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:473: in function 'for_each_tree'
	...im/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:175: in function 'prepare_highlight_states'
	...im/share/nvim/runtime/lua/vim/tree

**Comment by Create-Peace** [comment]
![image](https://github.com/user-attachments/assets/de68bdf1-46a1-4b83-bb0e-8ba6a6b4c3ec)
![image](https://github.com/user-attachments/assets/dc43ab47-63c4-4e87-b083-48f8b45f1231)

**Comment by folke** [comment]
`:TSUpdate`

**Comment by Create-Peace** [comment]
:TSUpdate, then 
![image](https://github.com/user-attachments/assets/5010cb31-80d7-4a50-8f92-e4f3afa0b677)

**Reply by dpetka2001** [reply]
Do you have the necessary compiler in Windows so that the Treesitter parsers can be compiled? If not, read through the Treesitter github repo and search for `Windows support` section for what you have to do.

**Reply by smithwinston** [reply]
I've seen this "Failed to execute the following command" errors with nvim+lazy on Windows too for tree-sitter parser updates.  I find if I just manually execute the `move` command in powershell it works just fine so it doesn't appear that there's a permissions issue -- I always seem to see it with Markdown parsers but then again I mostly do C/C++ so maybe it so far only the md parsers got updated.  The .dll does get built, it's the `move` from parser.dll over to ts-whatever.dll that is failing -- it's worth noting that I am also using Wezterm on Windows.

I wonder if it's possibly happening because the existing version of the .dll is in use by nvim and it can't overwrite it? (UNIX doesn't have that "file in use" problem).  Not really sure how to debug this further.

**Reply by Create-Peace** [reply]
@smithwinston Has this problem been resolved yet?

**Reply by Create-Peace** [reply]
I manually moved the corresponding .so files to the correct directory, and everything is working fine now.

**Reply by smithwinston** [reply]
I haven't seen that it was fixed, but until there's a tree-sitter update to the C or Rust grammars (which is what I primarily use), I won't know.  I think I last saw this issue in early January.

**Reply by smithwinston** [reply]
There was finally an update to a tree-sitter parser I use (c/cpp) so I could re-try this -- upon updating, I still see the same problem.   

The issue is that unlike UNIX, Windows won't allow the `python.so` to be copied over the destination one if the destination one is currently loaded.  So I think nvim-treesitter is going to have to either try to first unload the target parser DLL before copying over it, or I think that if you first rename the target to something else (e.g. `rename python.so python.old`) you can then copy in the new one with the correct name, however any running processes will still be using the old one (and you'll need to clean up that later).

The workaround is to simply quit neovim, start it up without any source files open (so no tree-sitter parser dll's loaded) and then do your `:TSUpdate` in which case there won't be any issues replacing the file.

**Comment by Create-Peace** [comment]
Another error has occurred.[Neo-tree ERROR] Error opening file: vim/_editor.lua:431: nvim_exec2(): Vim(edit):E576: Failed to parse ShaDa file: extra bytes in msgpack string at position 12269

**Reply by folke** [reply]
That's Neovim. Check the issues in the NEovim repo for things related to shada
